sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1058,101 @@ def _eval_is_nonnegative(self):
+    as the number of ways of depositing n distinct objects into m distinct bins,
+    with k1 objects in the first bin, k2 objects in the second bin, and so on.
+
+    For the sake of convenience for any negative integer `k` this function
+    will return zero.
+
+    To expand the multinomial when `k` contains a symbol, use either
+    ``expand_func()`` or ``expand(func=True)``. The former will keep
+    the polynomial in factored form while the latter will expand the
+    polynomial itself. See examples for details.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Rational, multinomial, expand_func, symbols, gamma
+    >>> n = Symbol('n', integer=True, positive=True)
+
+    >>> multinomial(15, 8)
+    490314
+
+    Comparing Coefficents generated with the multinomial function:
+
+    >>> a, b, c = symbols('a b c')
+    >>> ((a+b+c)**3).expand(func=True)
+    a**3 + 3*a**2*b + 3*a**2*c + 3*a*b**2 + 6*a*b*c + 3*a*c**2 + b**3 + 3*b**2*c + 3*b*c**2 + c**3
+
+    The Coefficient when of 2 1 0 corresponding to the second term is 3
+
+    >>> multinomial(2, 1, 0)
+    3
+
+    >>> multinomial(Rational(5, 4), 3)
+    gamma(21/4)/(6*gamma(9/4))
+
+    >>> multinomial(n, 3).rewrite(gamma)
+    gamma(n + 4)/(6*gamma(n + 1))
+
+    >>> multinomial(n, 3).expand(func=True)
+    n**3/6 + n**2 + 11*n/6 + 1
+
+    >>> expand_func(multinomial(n, 3))
+    (n + 1)*(n + 2)*(n + 3)/6
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Multinomial_theorem#Multinomial_coefficients
+
+    .. [2] http://functions.wolfram.com/GammaBetaErf/Multinomial/02/0001/
+
+    """
+
+    @classmethod
+    def eval(self, *args):
+        args = list(map(sympify, args))
+        if len(args) < 2:
+            raise BadArgumentsError("A miniumum of 2 args required")
+        if any(x.is_Integer and x.is_negative for x in args):
+            return S.Zero
+        else:
+            if all(x.is_Integer for x in args):
+                from sympy import binomial
+                if len(args) == 2:
+                    return binomial(sum(args), args[1])
+                return Mul(*[binomial(j, k) for j, k in zip([Add(*args[:i]) for i in range(1, len(args) + 1)], args)])
+            elif all(x.is_number for x in args):
+                from sympy import gamma
+                return gamma(Add(*args) + 1)/Mul(*(gamma(x + 1) for x in args))
+
+    def _eval_rewrite_as_factorial(self, *args, **kwargs):
+        return factorial(Add(*args))/Mul(*(factorial(x) for x in args))
+
+    def _eval_expand_func(self, **hints):
+        from sympy import gamma, expand_func
+        return expand_func(gamma(Add(*self.args) + 1)/Mul(*(gamma(x + 1) for x in self.args)))
+
+    def _eval_rewrite_as_binomial(self, *args, **kwargs):
+        from sympy import binomial
+        return Mul(*[binomial(j, k) for j, k in zip([Add(*args[:i]) for i in range(1, len(args) + 1)], args)])
+
+    def _eval_rewrite_as_gamma(self, *k, **kwargs):
```

Comment:
I'm not sure this should be here, either. If it was returned as an unevaluated multinomial it was because the assumptions couldn't be determined so the binomial or gamma writing could be selected. (And if it was written in terms of binomial, *that* expression will respond to the rewrite request.)

Reply:
(else a Piecewise solution could be returned but I am not sure that will be too useful)

Reply:
Ohhh you mean that ! ... well it's for the rewrite function ... just like the one binomial has.

Reply:
wouldn't this be useful ?
```python3
>>> a = multinomial(n, k)
>>> a
multinomial(n, k)
>>> a.rewrite(binomial)
binomial(n, n)*binomial(k + n, k)
>>> a.rewrite(gamma)
gamma(k + n + 1)/(gamma(k + 1)*gamma(n + 1))
>>> a.rewrite(factorial)
factorial(k + n)/(factorial(k)*factorial(n))
>>> 
```

Reply:
per previous "oh wait" comment, the conditions for rewriting as gamma should be here, too. Perhaps it's better to rewrite as binomial and rewrite that as gamma -- and if it doesn't change from binomial form then it shouldn't be rewritten. Not sure.

Reply:
yes!, it can be rewritten as binomial and then as gamma but not the other way around, cause gamma doesn't rewrite to binomial.
that's one of the reasons i've added that here.

Reply:
Same comments as for binomial: if there are conditions that need to be met in order to write as gamma or binomial we have to check those conditions here before rewriting a symbolic multinomial in terms of gamma.